### PR TITLE
Update plugin to be compatible with 2021.3

### LINKED ISSUE #118: Compatibility with Intellij 2021.3
Intellij 2021.3 update screen is telling me that the OPA plugin wont be compatible with this version.

### PR DESCRIPTION
**Pull request by vgramer**
# Description
upgrade to gradle 7.2
make plugin compatible with idea 2021.3

Fixes #118 

```release-note
plugin compatibility with IDEA 2021.3 
```

# Special notes for your reviewer
some of the changes are not compatible with 211, so code is split into platform version code

update kotlin code:
  * Type mismatch: value of a nullable type T is used where non-nullable type is expected. This warning will become an error soon. See https://youtrack.jetbrains.com/issue/KT-36770 for details
  * 'toLong(): Long' is deprecated. Conversion of Char to Number is deprecated. Use Char.code property instead.

update idea skd call:
  * 'getVirtualFilesByName(Project!, String, GlobalSearchScope): (Mutable)Collection<VirtualFile!>' is deprecated. Deprecated in Java

### CODE REVIEW

File: gradle-211.properties [hunk 1]
```diff
@@ -10,7 +10,7 @@ psiViewerPluginVersion=211-SNAPSHOT
 
 # see https://www.jetbrains.org/intellij/sdk/docs/basics/getting_started/build_number_ranges.html for more information
 sinceBuild=211.6693
-untilBuild=212.*
+untilBuild=211.*
```
Comment (anderseknert) [inline gradle-211.properties:13]: Would it not be until 212.* as previously written?

File: src/212/main/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/go/time.kt [hunk 2]
```diff
@@ -0,0 +1,219 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+// IMPORTANT NOTE: this file is a  portage of time/format.go in kotlin
```
Comment (anderseknert) [inline src/212/main/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/go/time.kt:6]: Interesting. Why is this needed?

File: src/212/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationBase.kt [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+import com.intellij.openapi.util.io.FileUtil
+import com.intellij.util.ui.UIUtil
+import org.assertj.core.api.Assertions.assertThat
+import org.openpolicyagent.ideaplugin.OpaTestCase
+import org.openpolicyagent.ideaplugin.ide.runconfig.OpaConfigurationFactory
+import org.openpolicyagent.ideaplugin.ide.runconfig.RunConfigurationTestBase
+import java.nio.file.Path
+import java.nio.file.Paths
+
+abstract class OpaTestRunConfigurationBase : RunConfigurationTestBase() {
```
Comment (anderseknert) [inline src/212/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationBase.kt:26]: It's not 100% clear to me how this is related to the new version of IDEA.

File: src/212/main/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/go/time.kt (hunk 2 above)
Comment (vgramer) [reply src/212/main/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/go/time.kt:6]: in the first verion, for test runconfiguration i was parsing the the json output and time is express a long and the parsing with JDK didn't worked.  we currently parse the pretty output,  especially to have a nice explanation of failed test.  In the pretty output the time is express as string (number / unit) so I guess we won't need it anymore.

nice catch. i will create an issue to check if we still need it.

Comment (anderseknert) [reply src/212/main/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/go/time.kt:6]: Great! 😃

File: src/212/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationBase.kt (hunk 3 above)
Comment (anderseknert) [reply src/212/test/kotlin/org/openpolicyagent/ideaplugin/ide/runconfig/test/OpaTestRunConfigurationBase.kt:26]: Wow, quite some breaking change for a stdlib! But gotcha 👍

**Review by anderseknert — APPROVED** [review]

**Comment by ingvaldlorentzen** [comment]
As this is approved, any chance of getting this merged and a new release created soon? 

Sorry for the nagging, just really missing syntax highlighting of `.rego` in PyCharm 😅

**Comment by vgramer** [comment]
@ingvaldlorentzen  i will make a release this afternoon or tomorrow afternoon
